src/coordinator/ts/values.go

```diff
@@ -133,3 +139,39 @@ func NewFixedStepValues(millisPerStep time.Duration, numSteps int, initialValue
 		values:        values,
 	}
 }
+
+// At every resolution timestep, the last sample value is chosen
+func RawPointsToFixedStep(datapoints Datapoints, start time.Time, end time.Time, interval time.Duration) (FixedResolutionMutableValues, error) {
+	if end.Before(start) {
+		return nil, fmt.Errorf("start cannot be after end, start: %v, end: %v", start, end)
+	}
+	numSteps := int(end.Sub(start) / interval)
```

Comment:
Also, is it appropriate to always round down? Maybe have an option to take the ceil instead?

Reply:
good point, I think we make a lot of these decisions right now. I kept it same as how prometheus handles it for now. In the future, if we need that option, we can change the function to take in start, numSteps instead of end. But for now, doesn't seem likely we need that.